.github/workflows/main.yml

```diff
@@ -72,6 +72,32 @@ jobs:
         uses: ./.github/actions/set-up-notebook-testing
       - name: nb-tester tests
         run: tox -e nb-tester
+      - name: Check all notebooks are listed in config file
```

Comment:
Now that this test is a direct workflow, we could move the Python tests to be in the API check. That should speed up CI for content-only changes because we don't need to run tox. It's fine if that is changed in a follow up.

Also, I'd recommend this script be a dedicated file in `scripts/`. Even though it's pretty simple, it is more maintainable and discoverable living in a dedicated Python file. It keeps this YAML file simpler and allows us to do things like run Ruff on the file.

You could have a folder called `scripts/ci`?